frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1285,8 +1294,11 @@ impl<T: Config> Pallet<T> {
 	///
 	/// Returns `Ok(consumed_weight)` if operation is okay.
 	pub fn create_snapshot() -> Result<Weight, ElectionError> {
+		// we don't impose any limits on the targets for now, the assumption is that
+		// `T::DataProvider` will sensibly return small values to use.
 		let target_limit = <CompactTargetIndexOf<T>>::max_value().saturated_into::<usize>();
```

Comment:
Is it correct to assume we are not too worried about a target limit for now because we will set `MaxValidatorsCount` which should be reasonable to read in a single block?

Reply:
I also see we always fail if the target limit is less than the count of validators; if we ever are worried about a lot of validators creating memory issues but don't want to limit the amount waiting, we could apply the bags approach: we sort validators by self stake and `TargetsBagsList::iter().take(target_limit)`

Reply:
> Is it correct to assume we are not too worried about a target limit for now because we will set MaxValidatorsCount which should be reasonable to read in a single block?

Yup, and typically there are way less interest in being a validators than nominator.

> we sort validators by self stake and TargetsBagsList::iter().take(target_limit)

Indeed, we need a new instance of the `VoterList` that only contains validators.